### PR TITLE
fix(vscode): apply trace.server live, drop stale rslint.json watcher, port unicode-bom E2E

### LINKED ISSUE #39: Changing rstack.rslint.trace.server restarts every stack instead of updating trace in place
## Problem

Changing `rstack.rslint.trace.server` restarts the whole extension instead of updating the trace level in place.

`trace.server` is listed in the Rslint controller's `restartOnSettings` (`packages/vscode/src/stacks/lint/index.ts:71-75`). The shell answers any matching configuration change with `restart(undefined, …)` (`packages/vscode/src/extension.ts:73-110`), which expands to every stack (`extension.ts:248-257`). So flipping a debug-only setting from `off` to `messages`:

1. retires Rslint, Rstest and fmt (status bar goes to `starting`),
2. sends `didClose` for every open document and deletes its diagnostics,
3. disposes the Rstest `TestController` (the Test Explorer tree is destroyed and rebuilt),
4. stops every fmt server,
5. re-runs detection and rebuilds all three stacks.

The replacement client only attaches a trace channel and calls `client.setTrace` when the setting is non-`off` at construction time (`packages/vscode/src/stacks/lint/Rslint.ts:415-445,509-514`), so there is no path to enable tracing on a running client.

## Upstream behavior

The standalone rslint extension supplied a `traceOutputChannel` unconditionally and let `vscode-languageclient` own the setting: it puts the initial value into the `initialize` request's `trace` field and sends `$/setTrace` on every later change, with no restart (`rslint@origin/main:packages/vscode-extension/src/Rslint.ts:96-119,429-434`). Its E2E asserts that `off → messages → verbose → off` reaches every running client via `$/setTrace` while each server's start count stays at 1 (`packages/vscode-extension/__tests__/suite-jsconfig/trace-output-channel.test.ts:200-300`).

## Proposal

- Drop `trace.server` from `restartOnSettings` in `stacks/lint/index.ts`.
- Pass the trace channel to the language client unconditionally so `vscode-languageclient` handles `$/setTrace` itself; remove the manual `setTrace` call.
- Keep the shared `Rstack: Rslint` channel for trace output (the four-channel layout in `channels.ts` is a deliberate cap and unrelated to whether a restart is needed).
- Add an E2E in `e2e/lint/` mirroring the upstream assertions: changing the setting notifies running clients and does not restart any server.
- Check whether `rstack.fmt.trace.server` has the same issue and apply the same fix.

## Context

Found while auditing the standalone rslint/rstest extensions for sunset. This is the only confirmed behavior regression relative to the standalone extensions.

### LINKED ISSUE #41: Remove the stale rslint.json / rslint.jsonc entries from the lint config watcher
## Problem

`RSLINT_CONFIG_WATCH_NAMES` in `packages/vscode/src/stacks/lint/Rslint.ts:64-81` still lists `rslint.json` and `rslint.jsonc`, with a comment claiming "the Go server does load `rslint.json` from its cwd as a no-JS-config fallback in automatic mode".

That claim is stale. Current upstream automatic discovery only knows the four JS/TS module names and explicit loading rejects non-module extensions (`rslint@origin/main:internal/config/discovery/catalog.go:18-37`); the standalone extension has already removed the JSON names from its own watcher (`packages/vscode-extension/src/Rslint.ts:45-56`); and this extension's own detection deliberately ignores JSON (`src/detection.ts:25-29`, proven by `e2e/lint/suite-noconfig/noconfig.test.ts:181-192`).

## Observable effect

- Folder with only `rslint.json`: nothing happens, no runtime exists so no watcher is installed.
- Folder that already has a lint runtime (native `rslint.config.*` or an Rstack bridge): touching `rslint.json` is classified as `config-change`, debounced 300 ms and sent as a config-refresh (`Rslint.ts:124-130,516-520,582-620`). Go rebuilds the current config transaction with `loadMode: 'fresh'`, invalidates diagnostic and lint-project caches, re-evaluates the real JS/TS config and republishes every diagnostic (`internal/lsp/config_discovery.go:323-437,820-848`). No process restarts and the result is normally identical, so users are unlikely to notice, but it is a pointless full config re-evaluation triggered by a file that is never configuration.

## Proposal

Remove `rslint.json` / `rslint.jsonc` from `RSLINT_CONFIG_WATCH_NAMES` and delete the comment. No behavior change for real configs.

### LINKED ISSUE #42: Port the upstream suite-unicode-bom Rslint E2E
## Problem

The standalone rslint extension's `suite-unicode-bom` E2E was never ported. Every other functional upstream suite has a counterpart under `packages/vscode/e2e/lint/`; this one is missing, so the worker/proxy layer this extension adds between VS Code and the Go server has no coverage for the BOM boundary case.

Upstream source: `rslint@origin/main:packages/vscode-extension/__tests__/suite-unicode-bom/unicode-bom.test.ts` and `__tests__/fixtures-unicode-bom/`. Added in web-infra-dev/rslint#1560 (port of the `unicode-bom` rule, 2026-08-06), fixture moved from `rslint.json` to a native config in web-infra-dev/rslint#1951. rstackjs/rstack-editor#26 ported #1611 and #1279 but skipped #1560.

## What the upstream test guards

VS Code consumes a UTF-8 BOM as encoding metadata, so the document model sent over LSP contains no U+FEFF even though the file bytes start with `EF BB BF`. The `unicode-bom` rule once emitted a BOM-removal fix with range `[-1, 0]`, which cannot be expressed as a VS Code `TextEdit`; the final fix skips the rule for editor linting and leaves BOM fixes to the CLI (`internal/lsp/lint_generation.go:20-39`). The test asserts:

1. disk starts with `EF BB BF`, `TextDocument.getText()` has no U+FEFF;
2. diagnostics contain `no-var` (positive control) and nothing from `unicode-bom`;
3. whole-document Quick Fixes contain `Disable no-var for this line` and no BOM-related action;
4. document text and dirty state are unchanged.

## Why it still matters here

Static reading says this extension is transparent: the router only gates by runtime ownership (`stacks/lint/WorkspaceDocumentRouter.ts:101-176`), `vscode-languageclient` serializes `getText()` and forwards UTF-16 incremental ranges unchanged, and the worker forwards requests without rewriting text or positions (`stacks/lint/worker/index.ts:109-166,219-241`). The existing emoji test (`e2e/lint/suite/extension.test.ts:388-448`) proves UTF-16 incremental alignment, not BOM handling. That is reasoning, not evidence; E2E is the ground truth for editor behavior in this repo.

## Proposal

Port `suite-unicode-bom` and its fixture into `packages/vscode/e2e/lint/` as-is (native `rslint.config.mjs` enabling `unicode-bom` and `no-var`, per the post-#1951 shape), register it in `e2e/run.mjs`, and run `pnpm test:e2e:lint`.

### PR DESCRIPTION
**Pull request by fi3ework**
## Summary

- Apply `rstack.rslint.trace.server` live through `vscode-languageclient`, without restarting the language server.
- Scope `rstack.rslint.trace.server` to the window, matching `vscode-languageclient`'s URI-less configuration lookup and the existing `rstack.fmt.trace.server` scope.
- Remove the stale `rslint.json` / `rslint.jsonc` config-refresh watcher entries now that Rslint no longer loads legacy JSON configs.
- Port Rslint's Unicode BOM E2E coverage, preserving this extension's diagnostic-code adaptation.

Upstream provenance: [rslint #1967](https://github.com/web-infra-dev/rslint/pull/1967) for live LSP tracing, [rslint #1951](https://github.com/web-infra-dev/rslint/pull/1951) for removing legacy JSON config loading, and [rslint #1560](https://github.com/web-infra-dev/rslint/pull/1560) for the Unicode BOM rule and editor coverage.

The language client id changes from `rslint` to `rstack.rslint` because `vscode-languageclient` derives the trace setting namespace from that id. Without the rename, it would watch the unsupported `rslint.trace.server` key instead of `rstack.rslint.trace.server`.

## Related Links

Closes #39

Closes #41

Closes #42

## Checklist

- [x] Tests updated (or not required).
- [x] Documentation updated (or not required).

## Testing

- `pnpm lint` — exit 0: `Found 0 lint errors, 0 type errors and 0 warnings (linted 165 files with 22 rules, type-checked 156 files in 290ms using 14 threads)`.
- `pnpm test:unit` — exit 0: 31 test files passed and 264 tests passed.
- `RSTACK_LINT_E2E_SUITES="Trace output,unicode-bom,No config,JS config" VSCODE_CLI=1 pnpm test:e2e lint` — exit 0: JS config, Trace output channel, No config, and unicode-bom suites all passed.
- `Rstack lint bridge` suite: covered by CI after the rebase onto main (fixed there by #46).

### CODE REVIEW

File: packages/vscode/src/stacks/lint/Rslint.ts [hunk 1]
```diff
@@ -440,12 +434,8 @@ export class Rslint implements Disposable {
         );
       },
     };
-    if (traceEnabled) {
-      clientOptions.traceOutputChannel = this.lspOutputChannel;
-    }
-
     const client = new ManagedLanguageClient(
-      'rslint',
+      'rstack.rslint',
```
[the remark this answers is not in the document]
Comment (fi3ework) [reply packages/vscode/src/stacks/lint/Rslint.ts:439]: Agreed — vscode-languageclient reads `trace.server` without a resource URI. Fixed in [`0627884`](https://github.com/rstackjs/rstack-editor/commit/0627884acaa693fbba5925f2bc31e00cd4c13069): `rstack.rslint.trace.server` is now window-scoped like `rstack.fmt.trace.server`, with a unit test pinning both scopes.
